libavformat/mpegtsenc.c

```diff
@@ -1551,9 +1551,24 @@ static void mpegts_write_pes(AVFormatContext *s, AVStream *st,
     ts_st->prev_payload_key = key;
 }
 
+static int is_h264_start_code_3bytes(const uint8_t* p) 
+{
+    if (p[0] == 0x0 && p[1] == 0x00 && p[2] == 0x01) return 1;
+    return 0;
+}
+
+static int can_find_h264_start_code(const uint8_t* p, size_t size) 
+{
+    for (int i = 0; i <= (size - 3); i++) {
```

Comment:
couldn't we cut to the chase and check only offsets 0 [`ts`] and 5 [`loomts`]?

Reply:
Yeah, that would work.

I find it cleaner this way, where we inspect rigorously the stream, in a general way. We would need some documentation on why we inspect that specific offset